README.md

```diff
@@ -0,0 +1,219 @@
+
+    development:
+      secret_key_base: kaleisgreat  # probably not the most secure key ever
+
+Somewhere else, Rails has a method, let's call it `sign`, which takes a `message` and a `key` and returns a `signature`, which is just a string:
+
+    # sign(message: string, key: string) -> signature: string
+    def sign(message, key):
+      # cryptographic magic here
+      return signature
+    end
+
+It's guaranteed that given the same message and key, sign will produce output. Also, without the key, it is practically impossible to know what `sign` would return for a given message. That is, signatures can't be forged.
+
+Rails creates a signature for every cookie it sets, and appends the signature to the cookie.
+
+When it receives a cookie, Rails verifies that the signature matches the content (that is, that `sign(cookie_body, secret_key_base) == cookie_signature`).
+
+This prevents cookie tampering. If a user tries to edit their cookie and change the `cart_id`, they signature won't match, and Rails will silently ignore the cookie and set a new one.
+
+# Tying it together
+
+In our `items_controller.rb`, we might have an `add_to_cart` method, which is called
+when the user adds something to their cart. It might work something like this:
+
+    # Routed from POST /items/:id/add_to_cart
+    def add_to_cart
+      # Get the item from the path
+      @item = Item.find(params[:id])
+      
+      # Load the cart from the session, or create a new empty cart.
+      cart = session[:cart] || []
+      cart << @item.id
+
+      # Save the cart in the session.
+      session[:cart] = cart
+    end
+
+That's it! It's common to wrap this up in a helper method:
+
+It's common to wrap up some of this functionality in a helper method:
+
+    class ApplicationController < ActionController::Base
+      def current_cart
+        session[:cart] ||= []
+      end
+    end
+
+So now our controller looks like this:
+
+    # Routed from POST /items/:id/add_to_cart
+    def add_to_cart
+      # Get the item from the path
+      @item = Item.find(params[:id])
+      
+      # Load the cart from the session, or create a new empty cart.
+      current_cart << @item.id
+    end
+
+This way, we can use `current_cart` in our views and layouts too. For example, we may want to show the user how many items are in their cart as part of the layout.
+
+## Conclusion
+
+Cookies are foundational for the modern web.
+
+You've probably heard a lot about cookies from non-technical sources. In fact, the EU recently passed a law requiring sites that use cookies to request user consent before doing so.
+
+The effect of this law was not unlike the effect of California's [Proposition 65][prop_65], which states that buildings must have clear signage if they contain any one of a list of chemicals known to the state of California to cause cancer.
```

Comment:
I don't want to bring politics into this as much as possible. Just skip to "most site use cookies..."